### ISSUE TITLE
GitHub Action fails to configure AWS credentials when PR is cut from fork

### ISSUE THREAD

**Issue by trivikr** [aws/aws-sdk-js-v3#1267]
**Describe the bug**
GitHub Action fails to configure AWS credentials when PR is cut from fork

Examples:
* https://github.com/aws/aws-sdk-js-v3/pull/1266
* https://github.com/aws/aws-sdk-js-v3/pull/1257

**To Reproduce (observed behavior)**
* Create a PR with aws/aws-sdk-js-v3 from your fork
* Notice that GitHub Action CI build fails with "Could not load credentials from any providers"

**Expected behavior**
The CI should be able to fetch AWS credentials, and proceed with running the tests.

**Screenshots**
<details>
<summary>Screenshot</summary>

![Screen Shot 2020-06-15 at 5 31 02 PM](https://user-images.githubusercontent.com/16024985/84718676-17bf7080-af2e-11ea-944f-821921c24097.png)

</details>

**Additional context**
* The GitHub Action to run CI was introduced in https://github.com/aws/aws-sdk-js-v3/pull/1265
* This is a limitation from GitHub Secrets https://github.com/aws-actions/aws-codebuild-run-build?files=1#credentials-and-permissions

**Comment by trivikr** [comment]
Temporary solution to unblock PRs, is to create PRs from branch in upstream instead of fork.
* Example PR from fork branch: https://github.com/aws/aws-sdk-js-v3/pull/1266
* Same PR with upstream branch: https://github.com/aws/aws-sdk-js-v3/pull/1268

**Comment by trivikr** [comment]
There's an open request on GitHub Community to allow secrets to be shared with forks from trusted Actions https://github.community/t/allow-secrets-to-be-shared-with-forks-from-trusted-actions/16525

GitHub has not responded on the request though.

**Comment by trivikr** [comment]
GitHub staff clarified that they currently don't have plans to make secrets available to builds of forks
Details: https://github.community/t/make-secrets-available-to-builds-of-forks/16166/14

**Comment by trivikr** [comment]
There's a hack to share secrets with forks, but it requires comment from a developer with write permission to start the CI https://github.com/imjohnbo/ok-to-test

**Comment by trivikr** [comment]
One solution is to use CodeBuild on push, and run unit tests on GitHub Action.

Cons: [GitHub Actions job runners](https://help.github.com/en/actions/reference/virtual-environments-for-github-hosted-runners#supported-runners-and-hardware-resources) have 2 x86_64 CPU cores and 7 GB RAM.
The CI speed can be improved by running tests for only changes packages.

**Comment by trivikr** [comment]
The following command will build only the packages which are changed, and their dependents:
`lerna run build --since master --include-dependents`

The following command will run test for only changed packages:
`lerna run test --since master`

**Comment by trivikr** [comment]
We're reverting CI which uses AWS credentials from GitHub Secrets as of now https://github.com/aws/aws-sdk-js-v3/pull/1278

We'll re-introduce GitHub Actions without CodeBuild like attempted in #1272 after moving to root TSConfig with compilerOptions.paths and ts-jest.
